project_access_tokens.go

```diff
@@ -0,0 +1,141 @@
+//
+// Copyright 2021, Patrick Webster
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gitlab
+
+import (
+	"fmt"
+	"time"
+	"net/http"
+
+)
+
+// ProjectAccessTokensService handles communication with the
+// project access tokens related methods of the GitLab API.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/resource_access_tokens.html
+type ProjectAccessTokensService struct {
+	client *Client
+}
+
+// ProjectAccessToken represents a GitLab Project Access Token.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/resource_access_tokens.html
+type ProjectAccessToken struct {
+	ID        	int   `json:"id"`
+	UserID      int   `json:"user_id"`
+	Name        string   `json:"name"`
+	Scopes      []string       `json:"scopes"`
+	ExpiresAt   *time.Time 	 `json:"expires_at"`
+	CreatedAt   *time.Time              `json:"created_at"`
+	Active      bool              `json:"active"`
+	Revoked    bool            `json:"revoked"`
+	Token  string            `json:"token"`
+}
```

Comment:
The code should be gofmt'ed. This struct looks ugly

Reply:
I formatted the code and added some logic to properly handle the timestamps GitLab returns when getting a token.